consensus/src/dag/dag_handler.rs

```diff
@@ -76,19 +76,45 @@ impl NetworkHandler {
                     }
                 },
                 Some(res) = self.node_fetch_waiter.next() => {
-                    if let Err(e) = res.map_err(|e| anyhow::anyhow!("recv error: {}", e)).and_then(|node| self.node_receiver.process(node)) {
+                    match res {
+                        Ok(node) => if let Err(e) = self.node_receiver.process(node).await {
                         warn!(error = ?e, "error processing node fetch notification");
-                    }
+                        },
+                        Err(e) => {
+                            debug!("sender dropped channel: {}", e);
+                        },
+                    };
                 },
                 Some(res) = self.certified_node_fetch_waiter.next() => {
-                    if let Err(e) = res.map_err(|e| anyhow::anyhow!("recv error: {}", e)).and_then(|certified_node| self.dag_driver.process(certified_node)) {
-                        warn!(error = ?e, "error processing certified node fetch notification");
-                    }
+                    match res {
+                        Ok(certified_node) => if let Err(e) = self.dag_driver.process(certified_node).await {
+                            warn!(error = ?e, "error processing certified node fetch notification");                        },
+                        Err(e) => {
+                            debug!("sender dropped channel: {}", e);
+                        },
+                    };
                 }
             }
         }
     }
 
+    fn verify_incoming_rpc(&self, dag_message: &DAGMessage) -> Result<(), anyhow::Error> {
+        match dag_message {
+            DAGMessage::NodeMsg(node) => node.verify(&self.epoch_state.verifier),
+            DAGMessage::CertifiedNodeMsg(certified_node) => {
+                certified_node.verify(&self.epoch_state.verifier)
+            },
+            DAGMessage::FetchRequest(request) => request.verify(&self.epoch_state.verifier),
+            _ => {
+                error!(
+                    "unknown rpc message {:?}",
+                    std::mem::discriminant(dag_message)
+                );
+                Err(anyhow::anyhow!("unexpected rpc message"))
```

Comment:
we should avoid double logging, either we return error or we log and return nothing. I think for function like this we can just return the error back but in process_rpc should probably just log it and only propagate NeedSync status